File: week3/ex2.R
library(ggplot2)

setwd("/Users/cmdb/qb25-answers/week3/")

#2.2 Allele frequency spectrum
af_data <- read.table("AF.txt", header = TRUE)
af_plot <- ggplot(af_data , aes(x = Allele_Frequency)) +
  geom_histogram(bins = 11, fill = "orchid3", color = "black") +
  labs(
    title = "Allele Frequency Spectrum",
    x = "Allele Frequency",
    y = "Count"
  ) +
  theme_minimal()
ggsave("allele_freq.png", plot = af_plot, width = 8, height = 6, dpi = 300)
# The histogram displays a somewhat normal distribution with most allele frequencies around 0.4 this was surprising to me considering you would expect the histogram to skew to the left. 

#2.3 Read depth distribution
dp_data <- read.table("/Users/cmdb/qb25-answers/week3/DP.txt", header = TRUE)
dp_data$Read_Depth <- as.numeric(dp_data$Read_Depth)
read_plot <- ggplot(dp_data, aes(x = Read_Depth)) +
  geom_histogram(
    bins = 21,
    fill = "plum1",
    color = "white"
  ) +
  labs(
    title = "Read Depth Distribution Across All Samples",
    x = "Read Depth",
    y = "Number of Observations"
  ) +
  xlim(0, 20) +
  theme_minimal()
ggsave("read_depth.png", plot = read_plot, width = 8, height = 6, dpi = 300)

# The second histogram shows that most reads have lower depths between 5 and 15. This distribution looks like a right-skewed distribution, which is typical for sequencing data.

File: week3/ex2.py
#!/usr/bin/env python3
import sys

vcf_file = "/Users/cmdb/qb25-answers/week3/biallelic.vcf"
af_output = "AF.txt"
dp_output = "DP.txt"

with open(vcf_file) as vcf, open(af_output, 'w') as af_out, open(dp_output, 'w') as dp_out:
    af_out.write("Allele_Frequency\n")
    dp_out.write("Read_Depth\n")
    for line in open(vcf_file):
        if line.startswith('#'):
            continue
        fields = line.rstrip('\n').split('\t')
        info_field = fields[7]
        for entry in info_field.split(';'):
            if entry.startswith("AF="):
                af_values = entry.split('=')[1] # split AF value after = and select value after that 
                for af in af_values.split(','):  # split string into list ['0.12', '0.45']
                    af_out.write(f"{af}\n")
                break
        sample_fields = fields[9:]
    
        for sample in sample_fields:
            sample_values = sample.split(':')
            dp_values = sample_values[2]
            dp_out.write(f"{dp_values}\n")
            
            
# 2.2 The histogram displays a somewhat normal distrubution with most allele frequencies around 0.4 this was surprising to me considering you would expect the hisotgram to skew to the left. 
# 2.3 The second histogram shows that most reads have lower depths between 5 and 15. This distribution looks like a right-skewed distribution, which is typical for sequencing data.